src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/FrozenObjectNode.cs

```diff
@@ -81,7 +85,12 @@ protected override void OnMarked(NodeFactory factory)
 
         public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
         {
-            return comparer.Compare(((FrozenObjectNode)other)._field, _field);
+            var otherFrozenObjectNode = (FrozenObjectNode)other;
+            int result = comparer.Compare(otherFrozenObjectNode._owningType, _owningType);
+            if (result != 0)
+                return result;
+
+            return Comparer<int>.Default.Compare(otherFrozenObjectNode._allocationSiteId, _allocationSiteId);
```

Comment:
```suggestion
            return _allocationSiteId.CompareTo(otherFrozenObjectNode._allocationSiteId);
```
Nit: Shorter, also it is what other Compare methods typically use